docs/filesystem-pv-disks.md

```diff
@@ -0,0 +1,142 @@
+# Using file-system based PersistentVolumes as virtual disk backends
+
+## Introduction
+
+So far Kubernetes PersistentVolumes are remote file-systems attached to a pod at
+runtime. This is currently the only way of exposing remote or local storage to
+pods, through a file-system.
+Despite the fact that storage is exposed as file-systems to pod, KubeVirt has
+no mechanism to use these plain file-system based PVs to act as backends for
+virtual disks.
+
+
+## Goals
+
+This proposal is about adding a simple mechanism to allow using file-system
+based PersistentVolumes for storing file images backing virtual disks.
+
+The focus is primarily on providing the right cluster levels mechanics to enable
+future changes for improving usability, performance, or other characteristics.
+
+
+## Non-Goals
+
+- Provide a totally automatic solution
+- Cover snapshots explicitly
+
+
+## Use-case
+
+The primary use-case is to allow a VM to use PVs as backend storage for their
+virtual disks.
+
+
+## Design Overview
+
+Currently KubeVirt only supports [direct PV disks](direct-pv-disks.md).
+This feature works, by assigning PVCs backed by network block storage (currently
+only iSCSI) to a VM.
+
+The mechanism proposed in this proposal is leveraging Kubernetes to attach
+the remote storage as a file-system to a container, to use this file-system to
+store a disk image to act as a backend storage to the VM's disk.
+
+To keep the story compatability between the block and file-system storage, this
+proposal assumes that the file-system based storage will also just support a 1:1
+mapping between the virtual disk and volume.
+This aligns with the fact that a block volume can also just back a single
+virtual disk.
+
+
+## API
+
+The general API to use PersistentVolume claims as virtual disk backends was
+introduced with [direct PV proposal](direct-pv-disks.md).
```

Comment:
Is there still value in this "direct pv" approach? What we're doing there is essentially a hack that only works under some very controlled scenarios. 

If we don't have a strong argument for keeping the "direct pv" feature around, I'm in favor of deprecating that approach as soon as the file PVC integration is implemented.

Reply:
On the long run I'd like to reintroduce it, but we need ot get networking in place before it will really work.